### DISCUSSION TITLE
3D terrain visualization

### DISCUSSION THREAD

**Discussion by klokan** [maplibre/maplibre#158] [Cross-project feature proposals]
<img width="740" alt="Screen_Shot_2021-06-30_at_09 39 53 (1)" src="https://user-images.githubusercontent.com/59284/207888646-edcfb8b8-3f7e-4e0c-8768-a49ae736ac4c.png">

Visualization of the elevation from DEM (RGBA tiles) including drawing the tracks, labels and points.

Improvements related to 3d terrain rendering - improvements and bug-fixing in existing Web implementation, implementation in Native, documentation.

**Ideas / tickets towards this direction should be proposed in different CORE projects of MapLibre.**

Ported from https://roadmap.maplibre.org/c/90-3d-terrain-visualization

**Comment by wipfli** [comment]
This project was selected to become a Bounty Direction. You find a tracking issue at https://github.com/maplibre/maplibre/issues/189

**Comment by vicb** [comment]
Would be nice to have 3D geometries as part of this (i.e. https://github.com/maplibre/maplibre-gl-js/issues/644)

I have a flight tracking app and that would be super helpful.

**Reply by HarelM** [reply]
I totally agree, but as I just wrote in the linked issue I think the main problem is with the MVT format that doesn't support elevation, and the geojson is converted to this format before being presented, so the elevation data gets lost (I might be wrong, it was a shallow check I made).

**Reply by vicb** [reply]
If that only is the problem there should be an easy solution:

```
       message Feature {
// existing
                optional uint64 id = 1 [ default = 0 ];
                repeated uint32 tags = 2 [ packed = true ];
                optional GeomType type = 3 [ default = UNKNOWN ];
                repeated uint32 geometry = 4 [ packed = true ];
// additional
                repeated uint32 elevation = 5 [ packed = true ];
        }
```

You might to pick a different index than 5 to account for future extension of the format.

But probably this should be proposed as an extension to the spec, about time to have elevation there, don't you think?

**Reply by wipfli** [reply]
@nyurik started a discussion on the future of the mvt specification. Feel free to add this suggestion there, @vicb: https://github.com/nyurik/future-mvt/discussions/1
